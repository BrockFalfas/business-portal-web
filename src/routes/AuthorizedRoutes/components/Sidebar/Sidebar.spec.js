import React from 'react';
import { shallow } from 'enzyme';

import { Sidebar, getSelectedKeyFromPath } from './index';

const initSidebar = overrides => {
  const mockProps = {
    logout: jest.fn(),
  };
  const wrapper = shallow(<Sidebar {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('routes: <Sidebar/>', () => {
  it('should render without crashing', () => {
    const { wrapper } = initSidebar();
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleLogout', () => {
    it('should call the logout function from props', () => {
      const { mockProps, wrapper } = initSidebar();

      wrapper.instance().handleLogout();

      expect(mockProps.logout).toHaveBeenCalled();
    });
  });

  describe('hovering with mouse', () => {
    it('should change collapsed state on mouse over/out', () => {
      const { wrapper } = initSidebar();
      expect(wrapper.state('collapsed')).toBe(true);

      wrapper.instance().handleMouseOver();
      expect(wrapper.state('collapsed')).toBe(false);

      wrapper.instance().handleMouseOut();
      expect(wrapper.state('collapsed')).toBe(true);
    });
  });

  describe('selected menu item', () => {
    it('should select the active menu item based on current route', () => {
      const { wrapper } = initSidebar({ pathname: '/contractors' });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('getSelectedKeyFromPath', () => {
    it('should return the first part of the path, with a leading slash', () => {
      expect(getSelectedKeyFromPath('/some/path/to/page')).toBe('/some');
      expect(getSelectedKeyFromPath('/some/')).toBe('/some');
      expect(getSelectedKeyFromPath('/some')).toBe('/some');
      expect(getSelectedKeyFromPath('/')).toBe('/');
    });

    it('should return an empty string on invalid routes', () => {
      expect(getSelectedKeyFromPath('some/path')).toBe('');
      expect(getSelectedKeyFromPath('')).toBe('');
      expect(getSelectedKeyFromPath(null)).toBe('');
      expect(getSelectedKeyFromPath(undefined)).toBe('');
    });
  });
});
