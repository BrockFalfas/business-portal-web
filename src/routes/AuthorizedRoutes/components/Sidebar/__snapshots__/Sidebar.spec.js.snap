// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes: <Sidebar/> selected menu item should select the active menu item based on current route 1`] = `
<Header
  className="Sidebar"
>
  <Menu
    mode="horizontal"
    selectedKeys={
      Array [
        "/contractors",
      ]
    }
    theme="light"
  >
    <MenuItem
      className="Sidebar-header"
    >
      <div
        className="Sidebar-header-brand"
      >
        <img
          className="anticon"
          src="images/THOR-icon.png"
        />
      </div>
    </MenuItem>
    <MenuItem
      key="/welcome"
    >
      <Link
        replace={false}
        to="/welcome"
      >
        <Icon
          type="notification"
        />
        <span>
          Welcome!
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="/dashboard"
    >
      <Link
        replace={false}
        to="/dashboard"
      >
        <Icon
          type="home"
        />
        <span>
          Account Summary
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="/payments"
    >
      <Link
        replace={false}
        to="/payments"
      >
        <Icon
          type="table"
        />
        <span>
          Payments
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="/satisfaction-surveys"
    >
      <Link
        replace={false}
        to="/satisfaction-surveys"
      >
        <Icon
          type="smile"
        />
        <span>
          Satisfaction Surveys
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      className="Sidebar-item-logout"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <Icon
          type="logout"
        />
        <span>
          Logout
        </span>
      </a>
    </MenuItem>
  </Menu>
</Header>
`;

exports[`routes: <Sidebar/> should render without crashing 1`] = `
<Header
  className="Sidebar"
>
  <Menu
    mode="horizontal"
    selectedKeys={
      Array [
        "",
      ]
    }
    theme="light"
  >
    <MenuItem
      className="Sidebar-header"
    >
      <div
        className="Sidebar-header-brand"
      >
        <img
          className="anticon"
          src="images/THOR-icon.png"
        />
      </div>
    </MenuItem>
    <MenuItem
      key="/welcome"
    >
      <Link
        replace={false}
        to="/welcome"
      >
        <Icon
          type="notification"
        />
        <span>
          Welcome!
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="/dashboard"
    >
      <Link
        replace={false}
        to="/dashboard"
      >
        <Icon
          type="home"
        />
        <span>
          Account Summary
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="/payments"
    >
      <Link
        replace={false}
        to="/payments"
      >
        <Icon
          type="table"
        />
        <span>
          Payments
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      key="/satisfaction-surveys"
    >
      <Link
        replace={false}
        to="/satisfaction-surveys"
      >
        <Icon
          type="smile"
        />
        <span>
          Satisfaction Surveys
        </span>
      </Link>
    </MenuItem>
    <MenuItem
      className="Sidebar-item-logout"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        <Icon
          type="logout"
        />
        <span>
          Logout
        </span>
      </a>
    </MenuItem>
  </Menu>
</Header>
`;
